cpp/velox/shuffle/VeloxShuffleWriter.cc

```diff
@@ -1338,6 +1341,7 @@ arrow::Status VeloxShuffleWriter::splitFixedWidthValueBuffer(const velox::RowVec
       complexTypeData_[partitionId]->flush(&out);
       allBuffers.emplace_back(valueBuffer);
       complexTypeData_[partitionId] = nullptr;
+      arenas_[partitionId] = nullptr;
```

Comment:
Why we release complexTypeData_, but the arenas still occupy  memory?

Reply:
The variable complexTypeData_ is PrestoVectorSerializer and it doesn't own memory. All the memory used by PrestoVectorSerializer is managed by StreamArena.  StreamArena release its memory by deconstructing  memory::Allocation . Allocation (allocations_ and allocation_ of StreamArena) is deconstructed only when StreamArena is deconstructed.